### DISCUSSION TITLE
교육 과정 관련 개선 의견

### DISCUSSION THREAD

**Discussion by dhihm** [huijeong-kim/ministore#6] [Ideas]
실습하면서 나온 피드백 중 하나로, 처음 코드를 받있을 때 어떤 순서로 구현해야 할지 약간 막막힌 감이 있다는 의견들이 있었습니다. 초기 구현 관련 가이드 문서가 있으면 좋겠다는 의견이 있었고, 테스트를 그룹화?하여 단계별로 수행 되도록 하자는 의견이 있었는데 추가로 논의 해보면 좋겠습니다. 😁

**Comment by huijeong-kim** [comment]
문서는 러프한 형식으로라도 위키에 업데이트 해 보도록 할게요! 상세히 바꾸는 건 같이해여...!

근데 한 번의 과제에서 하나의 파일 내 테스트를 통과시키는 게 목표인데, 이 안에서도 그룹화가 필요하다는 말씀이실까요? 여기는 순서 같은 건 크게 의미 없어서 차례대로 통과시켜보면 되지 않을까 싶긴 한데요.

**Comment by hsung-yang** [comment]
앗 디스커션 늦게 올려서 죄송합니다
지금은 그냥 다 돌려서 통과하게 만들면 되겠지만 ministore 최종 코드가 완성되면 모든 코드에 todo가 있는 상태에서 초기진입한 분들이 뭐부터 해야할 지 헷갈릴 수도 있을 것 같습니다
그래서 전체 테스트를 cargo test로 돌리는 것보다는 dependency를 가질 수 있는 코드들부터 테스트를 할 수 있도록 cargo test 범위를 나누면 좋을 것 같습니다.
예를 들면 ministore_test_0.sh 에는 config, main, data_type, data_info 관련 테스트가 있고
ministore_test_1.sh 에는 0번 테스트 +block_device 관련 테스트가 있으며
ministore_test_2.sh 에는 1번 테스트 + cli 자체, (block_device + cli) 테스트가 있는 방식 같이 차례대로 test를 열어주는 방식이면 어떨까 생각했습니다!

**Reply by huijeong-kim** [reply]
아하 이해했습니다 ㅎㅎ

script 로 열어주는 방식은 각 스크립트에서 `cargo test block_device`식의 커맨드를 호출하면 괜찮을까요?! 아님 이 커맨드를 가이드에 넣어두기만 해도 충분할 것 같기도 하고요.

아님 테스트 모듈에 `#[ignore]` 해 두고 과제 설명에 이거 지우고 코드 작성하라고 가이드 하는 것도 좋아 보입니당

**Reply by hsung-yang** [reply]
넵넵 스크립트 안에서 테스트를 특정하면 될 것 같습니다 ㅎㅎ

**Reply by dhihm** [reply]
저는 일단 가이드 안에만 넣어도 될것 같긴 하네요 ㅋ

**Reply by huijeong-kim** [reply]
일단 위키에 넣어 두고 나중에(다 완성된 후에) 스크립트를 추가해 보는 걸루....>!!
